src/neuroconv/tools/nwb_helpers.py

```diff
@@ -47,8 +47,11 @@ def get_default_nwbfile_metadata() -> DeepDict:
 
 def make_nwbfile_from_metadata(metadata: dict) -> NWBFile:
     """Make NWBFile from available metadata."""
-    metadata = dict_deep_update(get_default_nwbfile_metadata(), metadata)
-    nwbfile_kwargs = metadata["NWBFile"]
+    nwbfile_kwargs = metadata.get("NWBFile") or dict()
+    if "session_description" not in nwbfile_kwargs:
+        nwbfile_kwargs.update(session_description="No description.")
+    if "identifier" not in nwbfile_kwargs:
+        nwbfile_kwargs.update(identifier=uuid.uuid4())
     if "Subject" in metadata:
```

Comment:
This will fail because `session_start_time` is missing. I don't think we should accept a metadata that does not contain an "NWBFile" key. This is defined in the schema for the metadata. "NWBFile" is a required key and "session_start_time" is required to be defined within it. The alternative is to use a filler session_start_time, but I would rather require it. See https://github.com/catalystneuro/neuroconv/blob/3474d05ab066db309fdf47c226666d1becec8235/src/neuroconv/schemas/base_metadata_schema.json#L8-L21

Reply:
We agreed a while back to never to use default session start times (technically we used to use `datetime(1970,1,1)` and that was not good so we stopped doing it)

However, the change in question here is the same both before and after this update; `get_default_nwbfile_metadata()` **does not** yield a metadata structure with the `session_start_time` field - it MUST be contained in the passed dictionary

What is true is that if we hit the second condition of `"NWBFile"` not being in `metadata` at all, that is true... So I suppose we should make two assertions here

a) `assert "NWBFile" in metadata`

and

(b) `assert "session_start_time" in metadata["NWBFile"]`

before doing any further argument extraction/auto-population + file creation